mesonbuild/minstall.py

```diff
@@ -441,6 +446,25 @@ def do_copyfile(self, from_file: str, to_file: str,
         append_to_log(self.lf, to_file)
         return True
 
+    def do_symlink(self, target: str, link: str, full_dst_dir: str) -> bool:
+        abs_target = target
+        if not os.path.isabs(target):
+            abs_target = os.path.join(full_dst_dir, target)
+        if not os.path.exists(abs_target):
+            raise RuntimeError(f'Tried to install symlink to missing file {abs_target}')
+        if os.path.exists(link):
+            link_target = os.readlink(link)
+            if link_target == target:
+                self.log(f'Symlink {link} to {target} already exists, skipping')
```

Comment:
I don't think it's necessary to skip it, since if it exists and points to the same location then deleting and recreating it doesn't cause any harm. This is a matter of semantics, except for the next bit...

In the case where they are not the same, I'd say we can still overwrite it with the target destination we want. Rationale being, that we would overwrite a regular file, so why not overwrite a symlink too? :D

Reply:
These two comments really simplified the function, thanks!